src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -196,10 +196,9 @@ trait EntityComponent {
 
       for {
         preExistingEntityRecs <- lookupEntitiesByNames(workspaceContext.workspaceId, entities.map(_.toReference))
-        _ <- deleteEntityAttributes(preExistingEntityRecs)
         savingEntityRecs <- insertNewEntities(workspaceContext, entities, preExistingEntityRecs).map(_ ++ preExistingEntityRecs)
         referencedAndSavingEntityRecs <- lookupNotYetLoadedReferences(workspaceContext, entities, savingEntityRecs).map(_ ++ savingEntityRecs)
-        _ <- insertAttributes(entities, referencedAndSavingEntityRecs)
+        _ <- upsertAttributes(entities, (savingEntityRecs ++ preExistingEntityRecs).distinct, referencedAndSavingEntityRecs)
```

Comment:
i've discovered that I don't _need_ it, but it makes the generated raw sql query a bit nicer. there may be dupes across `savingEntityRecs ++ preExistingEntityRecs` if we're saving an updated version of an entity that already exists, so the entityIds clause could be something like `where ea.owner_id in (1,2,3,1,2)`